microservices/first/utils/db_manager.py

```diff
@@ -123,10 +127,11 @@ def update_deployment_fields(
         :param deployment_id: The ID of the deployment to update.
         :param updates: A dictionary where keys are column names and values are
             the new values for those columns.
+        :throws KeyError if `updates` is missing necessary field
```

Comment:
Not all fields in `DEPLOYMENT_FIELDS` are necessary to update i.e. updating just leader status